### PR TITLE
Fixing aws installer modules for non arm specific sections

### PR DESCRIPTION
**Pull request by kelbrown20**
For versions 4.10+

Description: ARM in not supported yet on AWS UPI, Restricted networks, government regions, secret regions and china. Removing arm instances in these sections of the docs. 

Previews: 

- These sections will have `amd64` and `arm64` in `controlPlane.architecture`/ `compute.architecture`. Also includes `Machine types for arm64 architectures`
1. [Installing a cluster on AWS with customizations](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-customizations.html#installation-supported-aws-machine-types_installing-aws-customizations) 
2. [Installing a cluster on AWS with network customizations](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-network-customizations.html#installation-supported-aws-machine-types_installing-aws-network-customizations)
3. [Installing a cluster on AWS into an existing VPC](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-vpc.html)
4. [Installing a private cluster on AWS](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-private.html)

-  These sections will only have `amd64` for `controlPlane.architecture`/ `compute.architecture`  and `Machine types for x86_64 architectures` only

1.  [Installing a cluster on AWS in a restricted network with user-provisioned infrastructure](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-restricted-networks-aws.html#installation-supported-aws-machine-types_installing-restricted-networks-aws) 
2. [Installing a cluster on AWS into a government region](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-government-region.html)
3. [Installing a cluster on AWS into a Top Secret Region](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-secret-region.html#installation-supported-aws-machine-types_installing-aws-secret-region)
4. [Installing a cluster on AWS China](https://deploy-preview-43933--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-china.html#installation-supported-aws-machine-types_installing-aws-china-region)

### CODE REVIEW

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -57,10 +57,10 @@ ifeval::["{context}" == "installing-aws-customizations"]
 :aws:
 endif::[]
 ifeval::["{context}" == "installing-aws-government-region"]
-:aws:
+:aws-non-arm:
```
Comment (yselkowitz) [inline modules/installation-configuration-parameters.adoc:60]:
```suggestion
:aws:
:aws-govcloud:
```

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -57,10 +57,10 @@ ifeval::["{context}" == "installing-aws-customizations"]
...
 endif::[]
 ifeval::["{context}" == "installing-aws-secret-region"]
-:aws:
+:aws-non-arm:
```
Comment (yselkowitz) [inline modules/installation-configuration-parameters.adoc:63]:
```suggestion
:aws:
:aws-top-secret:
```

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -72,7 +72,7 @@ ifeval::["{context}" == "installing-aws-vpc"]
 :aws:
 endif::[]
 ifeval::["{context}" == "installing-restricted-networks-aws-installer-provisioned"]
-:aws:
+:aws-non-arm:
```
Comment (yselkowitz) [inline modules/installation-configuration-parameters.adoc:75]:
```suggestion
:aws:
:aws-disconnected:
```

File: modules/installation-supported-aws-machine-types.adoc [hunk 4]
```diff
@@ -466,3 +482,8 @@ The following Amazon Web Services (AWS) instance types are supported with {produ
 
 |===
 ====
+endif::[]
+
+ifeval::["{context}" == "installing-restricted-networks-aws.adoc"]
+:!arm-upi:
```
Comment (yselkowitz) [inline modules/installation-supported-aws-machine-types.adoc:488]: Where is this conditional defined or used?

File: modules/installation-supported-aws-machine-types.adoc [hunk 5]
```diff
@@ -8,6 +8,21 @@
...
+endif::[]
+ifeval::["{context}" == "installing-aws-government-region"]
+:arm:
+endif::[]
+ifeval::["{context}" == "installing-aws-secret-region"]
+:arm:
+endif::[]
+ifeval::["{context}" == "installing-aws-china-region"]
+:arm:
+endif::[]
```
Comment (yselkowitz) [inline modules/installation-supported-aws-machine-types.adoc:24]: Instead of this...

File: modules/installation-supported-aws-machine-types.adoc [hunk 6]
```diff
@@ -383,6 +398,7 @@ The following Amazon Web Services (AWS) instance types are supported with {produ
 |===
 ====
 
+ifndef::arm[]
```
Comment (yselkowitz) [inline modules/installation-supported-aws-machine-types.adoc:401]:
```suggestion
ifndef::aws-govcloud,aws-top-secret,aws-disconnected[]
```
etc.

**Review by yselkowitz — CHANGES_REQUESTED** [review]
It is possible that some of these situations will be added for ARM before others (e.g. disconnected, UPI), so instead of saying "non-arm" or "x86-only", maybe better to specify by name?  Suggestions follow, but such a change of approach might require additional changes.

File: modules/installation-configuration-parameters.adoc (hunk 1 above)
Comment (kelbrown20) [reply modules/installation-configuration-parameters.adoc:60]: @yselkowitz I was thinking that once something is supported (whether its UPI, or government) the `ifdevel` can just be switched from `aws-non-arm` to `aws`, or do you think the more specific names are more clear?

File: modules/installation-supported-aws-machine-types.adoc (hunk 4 above)
Comment (kelbrown20) [reply modules/installation-supported-aws-machine-types.adoc:488]: That was my mistake, I meant to change that to the conditional I used. Will fix

**Comment by kelbrown20** [comment]
@yselkowitz How do these updates look?

File: modules/installation-configuration-parameters.adoc [hunk 7]
```diff
@@ -57,11 +57,13 @@ ifeval::["{context}" == "installing-aws-customizations"]
 :aws:
 endif::[]
 ifeval::["{context}" == "installing-aws-government-region"]
-:aws:
+:aws-govcloud:
 endif::[]
+//AWS on arm is currently not supported in government regions
```
Comment (yselkowitz) [inline modules/installation-configuration-parameters.adoc:62]: Comment should be before, not after.

File: modules/installation-configuration-parameters.adoc [hunk 8]
```diff
@@ -57,11 +57,13 @@ ifeval::["{context}" == "installing-aws-customizations"]
...
 ifeval::["{context}" == "installing-aws-secret-region"]
-:aws:
+:aws-secret:
 endif::[]
+//AWS on arm is currently not supported in secret regions
```
Comment (yselkowitz) [inline modules/installation-configuration-parameters.adoc:66]: Comment should be before, not after.

File: modules/installation-configuration-parameters.adoc [hunk 9]
```diff
@@ -72,8 +74,9 @@ ifeval::["{context}" == "installing-aws-vpc"]
 :aws:
 endif::[]
 ifeval::["{context}" == "installing-restricted-networks-aws-installer-provisioned"]
-:aws:
+:aws-restricted:
 endif::[]
+//AWS on arm is currently not supported on restricted networks
```
Comment (yselkowitz) [inline modules/installation-configuration-parameters.adoc:79]: Comment should be before, not after.

File: modules/installation-configuration-parameters.adoc [hunk 10]
```diff
@@ -654,7 +657,7 @@ a|For example, `sshKey: ssh-ed25519 AAAA..`.
 
 |====
 
-ifdef::aws[]
+ifdef::aws,aws-govcloud,aws-secret,aws-restricted[]
```
Comment (yselkowitz) [inline modules/installation-configuration-parameters.adoc:660]: This part is fine, but above at lines 635 and 638 (`publish: Internal|External`) there is conditional docs that need to need to take these new options into account.

**Review by yselkowitz — CHANGES_REQUESTED** [review]

**Comment by kelbrown20** [comment]
@mjpytlak Hi Mike, PTAL at these updated comments! Thank you!

File: modules/installation-configuration-parameters.adoc [hunk 11]
```diff
@@ -56,11 +56,13 @@ endif::[]
 ifeval::["{context}" == "installing-aws-customizations"]
 :aws:
 endif::[]
+//Starting in 4.10, aws on arm64 is only supported for installation on custom, network custom, private clusters and VPC . This attribute excludes arm64 from installing on gov regions. When government regions are supported on arm64, change `aws-govcloud` to `aws`.
```
Comment (mjpytlak) [inline modules/installation-configuration-parameters.adoc:59]:
```suggestion
//Starting in 4.10, aws on arm64 is only supported for installation on custom, network custom, private clusters and VPC . This attribute excludes arm64 content from installing on gov regions. When government regions are supported on arm64, change `aws-govcloud` to `aws`.
```

**Comment by mjpytlak** [comment]
@kelbrown20 Thanks for adding the detailed comments. Very helpful for another writer who might be asked to make an update. I left one small suggestion behind.

**Comment by kelbrown20** [comment]
@aleskandro Hi Alessandro! Can you PTAL at this Arm doc PR for QE review? Thank you!

**Comment by aleskandro** [comment]
Thanks @kelbrown20 

/lgtm

**Pull request merged**